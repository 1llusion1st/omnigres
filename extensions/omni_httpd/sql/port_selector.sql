--- Force omni_httpd to select a port
BEGIN;
WITH listener AS (INSERT INTO omni_httpd.listeners (address, port) VALUES ('127.0.0.1', 0) RETURNING id),
     handler AS (INSERT INTO omni_httpd.handlers (query) VALUES ($$SELECT$$))
INSERT INTO omni_httpd.listeners_handlers (listener_id, handler_id)
SELECT listener.id, handler.id
FROM listener, handler;
DELETE FROM omni_httpd.configuration_reloads;
END;

CALL omni_httpd.wait_for_configuration_reloads(1);

-- Ensure port was updated
SELECT count(*) FROM omni_httpd.listeners WHERE port = 0;

-- TODO: Test request on a given port. Needs non-shell http client