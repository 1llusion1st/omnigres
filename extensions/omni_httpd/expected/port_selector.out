--- Force omni_httpd to select a port
begin;
with
    listener as (insert into omni_httpd.listeners (address, port) values ('127.0.0.1', 0) returning id),
    handler as (insert into omni_httpd.handlers (query) values ($$SELECT$$))
insert
into
    omni_httpd.listeners_handlers (listener_id, handler_id)
select
    listener.id,
    handler.id
from
    listener,
    handler;
ERROR:  WITH query "handler" does not have a RETURNING clause
LINE 12:     handler;
             ^
delete
from
    omni_httpd.configuration_reloads;
ERROR:  current transaction is aborted, commands ignored until end of transaction block
end;
call omni_httpd.wait_for_configuration_reloads(1);
-- Ensure port was updated
select
    count(*)
from
    omni_httpd.listeners
where
    port = 0;
 count 
-------
     0
(1 row)

-- TODO: Test request on a given port. Needs non-shell http client
